src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3217,6 +3217,19 @@ private Token ScanParameter()
                         scanning = false;
                         break;
 
+                    case '=':
+                        if (ExperimentalFeature.IsEnabled("PSNativeParameterParsing"))
+                        {
+                            sb.Append(c);
+                            scanning = false;
+                            sawColonAtEnd = true;
```

Comment:
By setting `sawColonAtEnd = true` here, the resulted `ParameterToken` will have the public property [UsedColon](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/parser/token.cs#L1337) set to `true`, which is incorrect for this publicly exposed API.

```
## an example output of the ParameterToken

ParameterName : foo=
UsedColon     : True
Text          : -foo=
TokenFlags    : None
Kind          : Parameter
HasError      : False
Extent        : -foo=
```

Reply:
Also, the code here doesn't consider `InCommandMode()` as it does in the `default` branch, which will result in different tokens when `-f=` is tokenized in the expression mode. But this is less concerned than the incorrect value for the public property `UseColon`.

Reply:
@daxian-dbw @SeeminglyScience thanks - I was trying to treat HasColon as a behavioral "no space between parameter and argument" (which likely wouldn't have considered `=` when it was created) rather than adding a new property. More disturbing to me is the `ValidateFromRemainingArguments` concern because it wasn't caught by our current validation. I'll certainly be adding that as a test (or 4) when this is ready.

all of my earlier attempts (without the parser change) couldn't handle the `-foo=${a}..b` case which is why i went down this path. I think I'll have to _not_ have the parser change and handle it only in the native parameter binder which should still handle the original issue of `git diff $a..$b` 

back to the drawing board - marking this back to WIP

Reply:
I found a different way to do this with @SeeminglyScience, so I've removed the work in progress. The new solution works with the `bat` example as desired (solving the `ValueFromRemainingArguments` mishandling). Essentially, now when we see an `=` as part of a parameter, we re-parse the input as a generic token rather than a parameter. This will certainly work better for native applications. 

I also added some additional tests for ValueFromRemainingArguments to be a little more exotic.